test/server/core/gameSteps/regroupWindow/RegroupPhase.spec.ts

```diff
@@ -0,0 +1,144 @@
+describe('Regroup phase', function() {
+    integration(function(contextRef) {
+        describe('Regroup phase', function() {
+            it('should let both players draw 2 cards, choose what to put into resources and ready all exhausted units.',
+                function () {
+                    contextRef.setupTest({
+                        phase: 'action',
+                        player1: {
+                            resources: ['smugglers-aid', 'atst', 'atst', 'atst'],
+                            deck: ['foundling', 'atst', 'cartel-spacer', 'atst'],
+                            hand: ['moment-of-peace', 'wroshyr-tree-tender', 'attack-pattern-delta'],
+                            groundArena: ['ardent-sympathizer', 'baze-malbus#temple-guardian'],
+                            spaceArena: ['alliance-xwing'],
+                        },
+                        player2: {
+                            resources: ['smugglers-aid', 'atst', 'atst', 'atst'],
+                            deck: ['pyke-sentinel', 'cartel-spacer', 'atst'],
+                            hand: ['scout-bike-pursuer'],
+                            groundArena: ['wampa'],
+                            base: { card: 'dagobah-swamp', damage: 0 },
+                            spaceArena: ['tieln-fighter'],
+                        }
+                    });
+
+                    const { context } = contextRef;
+
+                    // old hand & deck setup
+                    const oldHandPlayer1 = [];
+                    const oldHandPlayer2 = [];
+                    const oldDeckPlayer1 = [];
+                    const oldDeckPlayer2 = [];
+                    const oldResourcesPlayer1 = [];
+                    const oldResourcesPlayer2 = [];
+
+                    // we need to create a deep copy since otherwise the original array receives changes
+                    context.player1.hand.forEach((val) => oldHandPlayer1.push(val));
```

Comment:
You can consolidate each of these into one line with their definitions like this:

```typescript
const oldHandPlayer1 = [...context.player1.hand];
```

the `...` operator unpacks the array elements

Reply:
great will use ... instead